nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -779,6 +785,15 @@ ObstacleLayer::raytraceFreespace(
 
     unsigned int cell_raytrace_max_range = cellDistance(clearing_observation.raytrace_max_range_);
     unsigned int cell_raytrace_min_range = cellDistance(clearing_observation.raytrace_min_range_);
+
+    int64_t dx = int64_t(x1) - int64_t(x0);
+    int64_t dy = int64_t(y1) - int64_t(y0);
+    double observation_dist = std::hypot(dx, dy);
```

Comment:
This does it correctly, but you shouldn't implicit sign cast on `hypot` and we use modern C++ casting, not C-style casting